subprojects/build-cache-http/src/main/java/org/gradle/caching/http/HttpBuildCache.java

```diff
@@ -85,4 +87,16 @@ public HttpBuildCacheCredentials getCredentials() {
     public void credentials(Action<? super HttpBuildCacheCredentials> configuration) {
         configuration.execute(credentials);
     }
+
+    @Nullable
+    @Override
+    public String getDisplayName() {
+        return "HTTP";
+    }
+
+    @Nullable
+    @Override
+    public Map<String, String> getConfigDescription() {
+        return ImmutableMap.of("url", url == null ? "" : url.toString(), "authenticated", Boolean.valueOf(credentials.getUsername() != null && credentials.getPassword() != null).toString());
```

Comment:
Good catch. The impl now reflects that exposing a safe uri and covering that case for the 'authenticated' flag.